### ISSUE TITLE
Slices are not detected

### ISSUE THREAD

**Issue by barbrovessman** [FirelyTeam/firely-validator-api#261]
R5

When validating an instance that contains a required slice, the error below occurs.  Sushi and genonce do not give any errors.

The profile is on Appointment and the slice on participant:

```
* participant ^slicing.discriminator.type = #type  
* participant ^slicing.discriminator.path = "$this.actor.resolve()"        
    
* participant ^slicing.rules = #open
* participant contains healthcareservice 1..*
* participant[healthcareservice] ^short = "HealthcareServices involved in appointment"
* participant[healthcareservice].actor 1..1
* participant[healthcareservice].actor only Reference(HealthcareService) 
```

```
{
  "issue": [
    {
      "severity": {
        "value": "error"
      },
      "code": {
        "value": "invalid"
      },
      "details": {
        "coding": [
          {
            "system": {
              "value": "http://hl7.org/fhir/dotnet-api-operation-outcome"
            },
            "code": {
              "value": "1028"
            }
          },
          {
            "system": {
              "value": "http://fire.ly/dotnet-sdk-operation-outcome-structdef-reference"
            },
            "code": {
              "value": "Appointment(http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment).participant[healthcareservice]"
            }
          }
        ],
        "text": {
          "value": "Instance count is 0, which is not within the specified cardinality of 1..*"
        }
      },
      "location": [
        {
          "value": "Appointment.participant, element Appointment(http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment).participant[healthcareservice]"
        }
      ],
      "expression": [
        {
          "value": "Appointment.participant, element Appointment(http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment).participant[healthcareservice]"
        }
      ]
    },
    {
      "severity": {
        "value": "error"
      },
      "code": {
        "value": "invalid"
      },
      "details": {
        "coding": [
          {
            "system": {
              "value": "http://hl7.org/fhir/dotnet-api-operation-outcome"
            },
            "code": {
              "value": "1028"
            }
          },
          {
            "system": {
              "value": "http://fire.ly/dotnet-sdk-operation-outcome-structdef-reference"
            },
            "code": {
              "value": "Appointment(http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment).participant[healthcareservice]"
            }
          }
        ],
        "text": {
          "value": "Instance count is 0, which is not within the specified cardinality of 1..*"
        }
      },
      "location": [
        {
          "value": "Appointment.participant, element Appointment(http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment).participant[healthcareservice]"
        }
      ],
      "expression": [
        {
          "value": "Appointment.participant, element Appointment(http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment).participant[healthcareservice]"
        }
      ]
    }
  ]
}
```

**Comment by ewoutkramer** [comment]
Is it possible for you to include the instance you validated against + the StructureDefinition?

**Comment by barbrovessman** [comment]
Sure! The attached Appointment is part of a Parameters instance being validated, but hopefully this is enough to replicate the issue
[Appointment-15f86208-17c7-441e-adae-8797c5183c82.json](https://github.com/FirelyTeam/firely-validator-api/files/13978808/Appointment-15f86208-17c7-441e-adae-8797c5183c82.json)
[StructureDefinition-InvitationCommunicationRequest.json](https://github.com/FirelyTeam/firely-validator-api/files/13978822/StructureDefinition-InvitationCommunicationRequest.json)

**Comment by mmsmits** [comment]
Hi @barbrovessman 
I don't think you've sent the correct profile. InvitationCommunicationRequest seems to be of type `CommunicationRequest`.
Could you send the `http://bki.skane.se/invanartjanster/fhir/StructureDefinition/InvitationAppointment` StructureDefinition?

**Comment by barbrovessman** [comment]
Hi @mmsmits ,

This is the InvitationAppointment StructureDefinition:

[StructureDefinition-InvitationAppointment.json](https://github.com/user-attachments/files/18062103/StructureDefinition-InvitationAppointment.json)

BR, Barbro

**Comment by alexzautke** [comment]
Hi @barbrovessman,

in the latest version of the profile, I can see that the discriminator was switched from "this.actor.resolve()" to "$this.type". I'm assuming that this will work with the validator. Therefore I'm going to close the issue. If there are any issues outstanding pleaser re-open the ticket.
